waiter/src/waiter/handler.clj

```diff
@@ -1282,3 +1282,101 @@
         request))
     (catch Throwable th
       (utils/exception->response th request))))
+
+
+(defn get-params-from-query [query]
+  (let [split-query (str/split query #"&")] 
+      (let [vector-query (map #(str/split % #"=") split-query)]
+          (into {} vector-query))))
+          
+
+(defn- execute-signal
+  "Helper function to send signals to instances of a service.
+   An instance needs to be approved for killing by peers before an actual kill attempt is made.
+   The function stops and returns true when a successful kill is made.
+   Else, it terminates after we have exhausted all candidate instances to kill or when a kill attempt returns a non-truthy value."
+  ;; PEER-ACK UNUSED
+  ;; dont need threadpool
+  [notify-instance-killed-fn peers-acknowledged-eject-requests-fn allowed-to-manage-service?-fn scheduler populate-maintainer-chan! timeout-config
+   instance-id service-id signal-type correlation-id thread-pool response-chan]
+  (let [{:keys [eject-backoff-base-time-ms inter-kill-request-wait-time-ms max-eject-time-ms]} timeout-config]
+    (cid/with-correlation-id
+      correlation-id
+      (async/go
+        (try
+          ;; DONT KNOW IF WE NEED REASON AND RESULT MAP
+          (let [request-id (utils/unique-identifier) ; new unique identifier for this reservation request
+                reason-map-fn (fn [] {:cid correlation-id :reason :kill-instance :request-id request-id :time (t/now)})
+                result-map-fn (fn [status] {:cid correlation-id :request-id request-id :status status})]
+            (log/info "Attempting to send signal to " instance-id)
+              ;; MAKE OUR OWN TIMER FOR INSTANCES?
+              ;; (metrics/service-timer instance-id "kill-instance")
+              ;;loop [exclude-ids-set #{}]
+                ;;let [instance (service/get-rand-inst populate-maintainer-chan! service-id (reason-map-fn) exclude-ids-set inter-kill-request-wait-time-ms)]
+                ;; [instance (into {} [:id instance-id :service-id service-id])]
+                    (let [instance {:id instance-id :service-id service-id}]
```

Comment:
We should not be needing to make a map here, instance id and service ID will be passed explicitly to the scheduler instance.